Race condition in replay protection

CI for x/crypto bump in #241 fails with:

```
WARNING: DATA RACE

Write at 0x00c0004ca008 by goroutine 151:

  sync.(*WaitGroup).Wait()

      /home/travis/.gimme/versions/go1.13.linux.amd64/src/internal/race/race.go:41 +0xee

  github.com/pion/dtls/v2.TestReplayProtection()

      /home/travis/gopath/src/github.com/pion/dtls/replayprotection_test.go:104 +0xa65

  testing.tRunner()

      /home/travis/.gimme/versions/go1.13.linux.amd64/src/testing/testing.go:909 +0x199

Previous read at 0x00c0004ca008 by goroutine 33:

  sync.(*WaitGroup).Add()

      /home/travis/.gimme/versions/go1.13.linux.amd64/src/internal/race/race.go:37 +0x18d

  github.com/pion/dtls/v2.TestReplayProtection.func1()

      /home/travis/gopath/src/github.com/pion/dtls/replayprotection_test.go:43 +0xbc

Goroutine 151 (running) created at:

  testing.(*T).Run()

      /home/travis/.gimme/versions/go1.13.linux.amd64/src/testing/testing.go:960 +0x651

  testing.runTests.func1()

      /home/travis/.gimme/versions/go1.13.linux.amd64/src/testing/testing.go:1202 +0xa6

  testing.tRunner()

      /home/travis/.gimme/versions/go1.13.linux.amd64/src/testing/testing.go:909 +0x199

  testing.runTests()

      /home/travis/.gimme/versions/go1.13.linux.amd64/src/testing/testing.go:1200 +0x521

  testing.(*M).Run()

      /home/travis/.gimme/versions/go1.13.linux.amd64/src/testing/testing.go:1117 +0x2ff

  main.main()

      _testmain.go:418 +0x347

Goroutine 33 (running) created at:

  github.com/pion/dtls/v2.TestReplayProtection()

      /home/travis/gopath/src/github.com/pion/dtls/replayprotection_test.go:56 +0x448

  testing.tRunner()

      /home/travis/.gimme/versions/go1.13.linux.amd64/src/testing/testing.go:909 +0x199

==================

--- FAIL: TestReplayProtection (0.45s)

    testing.go:853: race detected during execution of test
```

Ping @at-wat

--- a/replayprotection_test.go
+++ b/replayprotection_test.go
@@ -1,9 +1,11 @@
 package dtls
 
 import (
+	"context"
 	"net"
 	"reflect"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
@@ -24,7 +26,17 @@ func TestReplayProtection(t *testing.T) {
 	c2, c3 := dpipe.Pipe()
 	conn := []net.Conn{c0, c1, c2, c3}
 
-	var wgReplays, wgRoutines sync.WaitGroup
+	var wgRoutines sync.WaitGroup
+	var cntReplays int32 = 1
+
+	ctxReplayDone, replayDone := context.WithCancel(context.Background())
+
+	replaySendDone := func() {
+		cnt := atomic.AddInt32(&cntReplays, -1)
+		if cnt == 0 {
+			replayDone()
+		}
+	}
 
 	replayer := func(ca, cb net.Conn) {
 		defer wgRoutines.Done()
@@ -40,9 +52,9 @@ func TestReplayProtection(t *testing.T) {
 				return
 			}
 
-			wgReplays.Add(1)
+			atomic.AddInt32(&cntReplays, 1)
 			go func() {
-				defer wgReplays.Done()
+				defer replaySendDone()
 				// Replay bit later
 				time.Sleep(time.Millisecond)
 				if _, werr := cb.Write(b[:n]); werr != nil {
@@ -67,7 +79,7 @@ func TestReplayProtection(t *testing.T) {
 		i := i
 		c := c
 		wgRoutines.Add(1)
-		wgReplays.Add(1) // Keep locked until the final message
+		atomic.AddInt32(&cntReplays, 1) // Keep locked until the final message
 		var lastMsgDone sync.Once
 		go func() {
 			defer wgRoutines.Done()
@@ -81,7 +93,7 @@ func TestReplayProtection(t *testing.T) {
 				if b[0] == numMsgs-1 {
 					// Final message received
 					lastMsgDone.Do(func() {
-						wgReplays.Done()
+						defer replaySendDone()
 					})
 				}
 			}
@@ -101,7 +113,9 @@ func TestReplayProtection(t *testing.T) {
 			return
 		}
 	}
-	wgReplays.Wait()
+
+	replaySendDone()
+	<-ctxReplayDone.Done()
 	time.Sleep(10 * time.Millisecond) // Ensure all replayed packets are sent
 
 	for i := 0; i < 4; i++ {

